workspace/src/localization/localization_py/localization_py/state_estimation.py

```diff
@@ -16,175 +16,226 @@
 import os
 
 
-#TODO: what does this do?
 ament_tools_root = os.path.join(os.path.dirname(__file__), '.')
 sys.path.insert(0, os.path.abspath(ament_tools_root))
 
-from KalmanFilter import KalmanFilter
+from EKF import EKF
+from particleFilter import particleFilter as PF
 from Chrono_coordinate_transfer import graph
+
 class StateEstimationNode(Node):
     def __init__(self):
         super().__init__('state_estimation_node')
         
-        self.declare_parameter('vis', False)
-        self.vis = self.get_parameter('vis').get_parameter_value().bool_value
+
+        # ROS PARAMETERS
+        self.use_sim_msg = self.get_parameter("use_sim_time").get_parameter_value().bool_value
+
+        self.declare_parameter('SE_mode', "GT")
+        self.SE_mode = self.get_parameter('SE_mode').get_parameter_value().string_value
+
 
         #update frequency of this node
         self.freq = 10.0
 
-        if(self.vis):
-            matplotlib.use("TKAgg")
-            #data that will be used by this class
-            #TODO: a lot of these variables aren't used, this data needs to be cleaned up
-            self.fig = plt.figure()
-            self.fig.suptitle('Kalman Filter', fontsize = 20)
         self.gps = ""
         self.groundTruth = ""
-        #self.imu = ""
         self.mag = ""
-        self.state = ""
-
-        #mag measurements
-        self.mag_x = 0
-        self.mag_y = 0
-        self.mag_z = 0
-        #lat, lon, ground truth of each
-        self.lat = 0
-        self.lon = 0
-        self.gt_lat = 0
-        self.gt_lon = 0
-
-        #x, y, kf predictions of each, history of each
+
+        if self.use_sim_msg:
```

Comment:
What's our solution here?

Reply:
Not sure what you mean here; if you are talking about switching from `chrono_ros_msgs` to standard ROS message types, I think that is for a later PR.

Reply:
hmm ok... an intermediary would be a good way to go for now but I don't have a ton of time atm. Might be a good first issue for someone else who wants to start getting involved...

Sure we can wait to merge this until then, but would be nice to have a timeline, as there are 2 other PR's that will be similar to this one for Path_Planning and Control, and I'd like to get those done soonish. Hope that's ok :)

Reply:
@AaronYoung5 I looked at this a little closer... and I am much less against the custom message types now. Do both `ChDriverInputs` and `VehicleInputs` (from art messages) exist due to the duplication you have discussed between the two sides of `Chrono_ROS_bridge` (as I think you have mentioned a few times)? Also is what you mean that if I just use the 'VehicleInputs' from 'art_msgs.msg' then it won't have problems parsing in either simulation or reality?

Reply:
@StefanCaldararu What changed that you aren't against custom messages now? I'm still unsure. And I think VehicleInputs and ChDriverInputs are identical, so at least for those messages, there won't be parsing problems ya. But issues come up when you specify the wrong message type in the subscriber constructor cause it won't receive messages from different message types. 

Ideally, the control stack is written for **reality**, and it's our job to make that reality stack run in simulation. Do you agree/does that make sense?

Reply:
I think the solution is for this node to only subscribe to what's used in reality, i.e. `art_msgs.VehicleInputs` (or whatever we decide). But ya, if `chrono_ros_msgs.ChDriverInput` is used in the constructor and `art_msgs.VehicleInputs` is used on the topic, it won't get any messages.

And okay, where is best?